libass/ass.c

```diff
@@ -1356,15 +1356,15 @@ void ass_lazy_track_init(ASS_Library *lib, ASS_Track *track)
             ass_msg(lib, MSGL_WARN,
                    "PlayResY undefined, setting to %d", track->PlayResY);
         } else if (!track->PlayResY) {
-            track->PlayResY = track->PlayResX * 3 / 4;
+            track->PlayResY = track->PlayResX * 3LL / 4;
             ass_msg(lib, MSGL_WARN,
                    "PlayResY undefined, setting to %d", track->PlayResY);
         } else if (!track->PlayResX && track->PlayResY == 1024) {
             track->PlayResX = 1280;
             ass_msg(lib, MSGL_WARN,
                    "PlayResX undefined, setting to %d", track->PlayResX);
         } else if (!track->PlayResX) {
-            track->PlayResX = track->PlayResY * 4 / 3;
+            track->PlayResX = track->PlayResY * 4LL / 3;
```

Comment:
Note that this does not fix the `int` overflow on assignment if the value exceeds `INT_MAX`. As I noted in the pull request comment, we need to decide what to do in that case.